Microsoft.Toolkit.Uwp.UI.Animations/ConnectedAnimations/ConnectedAnimationHelper.cs

```diff
@@ -0,0 +1,278 @@
+        /// which item to animate
+        /// </summary>
+        /// <param name="listViewBase">The <see cref="ListViewBase"/></param>
+        /// <param name="key">The connected animation key to register with the <see cref="ConnectedAnimationService"/></param>
+        /// <param name="elementName">The name of the element in the <see cref="DataTemplate"/> to animate</param>
+        public void RegisterListItem(ListViewBase listViewBase, string key, string elementName)
+        {
+            if (listViewBase == null || string.IsNullOrWhiteSpace(key) || string.IsNullOrWhiteSpace(elementName))
+            {
+                return;
+            }
+
+            var props = new ConnectedAnimationProperties()
+            {
+                Key = key,
+                IsListAnimation = true,
+                ElementName = elementName,
+                ListViewBase = listViewBase
+            };
+
+            _connectedAnimationsProps[key] = props;
+        }
+
+        /// <summary>
+        /// Unregisters the ListViewBase element items from participating in the connected animation
+        /// <seealso cref="RegisterListItem(ListViewBase, string, string)"/>
+        /// </summary>
+        /// <param name="listViewBase">The <see cref="ListViewBase"/></param>
+        /// <param name="key">The connected animation key to unregister</param>
+        public void RemoveListItem(ListViewBase listViewBase, string key)
+        {
+            if (listViewBase == null || string.IsNullOrWhiteSpace(key))
+            {
+                return;
+            }
+
+            _connectedAnimationsProps.Remove(key);
+        }
+
+        private void Frame_Navigating(object sender, Windows.UI.Xaml.Navigation.NavigatingCancelEventArgs e)
+        {
+            var parameter = e.Parameter != null && !(e.Parameter is string str && string.IsNullOrEmpty(str)) ? e.Parameter : null;
+
+            var cas = ConnectedAnimationService.GetForCurrentView();
+            foreach (var props in _connectedAnimationsProps.Values)
+            {
+                if (props.IsListAnimation && parameter != null && ApiInformationHelper.IsCreatorsUpdateOrAbove)
+                {
+                    props.ListViewBase.PrepareConnectedAnimation(props.Key, e.Parameter, props.ElementName);
+                }
+                else if (!props.IsListAnimation)
+                {
+                    cas.PrepareToAnimate(props.Key, props.Element);
+                }
+                else
+                {
+                    continue;
+                }
+
+                _previousPageConnectedAnimationProps[props.Key] = props;
+            }
+
+            _connectedAnimationsProps.Clear();
+            _coordinatedAnimationElements.Clear();
+        }
+
+        private void Frame_Navigated(object sender, Windows.UI.Xaml.Navigation.NavigationEventArgs e)
+        {
+            var navigatedPage = (sender as Frame).Content as Page;
```

Comment:
Use pattern matching.

Reply:
pattern matching in these cases causes a vs warning